### PR TITLE
47 user param in request

### PR DESCRIPTION
**Pull request by natkam**

### CODE REVIEW

**Comment by natkam** [comment]
todo: merge migrations

File: aqueduct/gateway/tests/utils/base.py [hunk 1]
```diff
@@ -95,7 +100,7 @@ def setUpClass(cls):
                     f"vLLM integration tests require vllm to be installed. "
                     f"Import error: {_VLLM_IMPORT_ERROR}"
                 )
-            print(f"\nStarting vLLM server for {cls.__name__}...")
+            logger.info("\nStarting vLLM server for %s...", cls.__name__)
```
Comment (meffmadd) [inline aqueduct/gateway/tests/utils/base.py]: Use prints for test related info to be able to differentiate application logs from test logs. Much easier to see and debug in CI etc!

File: aqueduct/gateway/views/decorators.py [hunk 2]
```diff
@@ -234,6 +287,22 @@ async def wrapper(request: ASGIRequest, *args, **kwargs):
...
+        try:
+            user_id = json.loads(request.body.decode("utf-8")).get("user_id", "")
+            return str(user_id)
+        except Exception as err:
+            log.info(f"Cound not retrieve user_id from request body: {err}")
+            return ""
+    else:
+        return ""
+
+
```
[the remark this answers is not in the document]
Comment (meffmadd) [reply aqueduct/gateway/views/decorators.py]: Also should the default for the user_id really be `""`? None would make more sense imo to differentiate between an acutally missing user_id vs. it being set to an empty string.

File: aqueduct/gateway/tests/test_endpoints.py [hunk 3]
```diff
@@ -186,15 +252,18 @@ def test_chat_completion_base64_file_input(self):
                 }
             ],
             "max_completion_tokens": 50,
+            "user_id": 42,
```
Comment (meffmadd) [inline aqueduct/gateway/tests/test_endpoints.py]: A test with email might be nice as well for the upcoming #71. It should work since spaces also work (`test_embeddings_with_user_id_in_body`) Also a separate test class for user_id might be helpful here as well since `test_chat_completion_base64_file_input` now also tests "user_id works with int value" and `test_embeddings_with_user_id_in_body` tests "user_id works with spaces". Gets quite confusing quickly. See other comment. This is the perfect candidate for a parameterized test with all "edge cases" in one place instead of it being scattered throughout different tests with names that describe something else.

File: aqueduct/gateway/tests/test_endpoints.py [hunk 4]
```diff
@@ -550,7 +608,7 @@ async def test_chat_completion_streaming(self):
         # Parse each chunk as JSON and collect content pieces
         content_pieces = _parse_streamed_content_pieces(streamed_lines)
         full_content = "".join(content_pieces).strip()
-        print(f"Full streamed content: {full_content}")
+        logger.debug("Full streamed content: %s", full_content)
```
[the remark this answers is not in the document]
Comment (natkam) [reply aqueduct/gateway/tests/test_endpoints.py]: That's not old-style string formatting, that's [`debug`'s signature](https://docs.python.org/3/library/logging.html#logging.Logger.debug) :grin: The first argument is *a format string*. Of course Python will not complain if you pass a string without any % placeholders.

Also - not that it matters much here - with this signature, the string will only be formatted if the log message actually gets logged. An f-string will always be formatted.

**Review by meffmadd — CHANGES_REQUESTED** [review]
